lib/OpenQA/Task/Job/Limit.pm

```diff
@@ -26,6 +27,7 @@ sub register {
 
 sub _limit {
     my ($job, $args) = @_;
+    enable_retry_on_signals $job;
```

Comment:
I was thinking a bit more about the API, and wouldn't this be nicer as an object? 🤔 
```perl
my $signal_guard = OpenQA::Task::SignalGuard->new($job, {retry => 1});
...
$signal_guard->retry(0);
...
```
On `DESTROY` it could also reset the signal handlers to the previous value. That would probably result in less unintended side effects when jobs run in foreground (without spawning a job process).

Reply:
Ok, I'll do that.